### PR TITLE
feat: update leaflet

### LINKED ISSUE #877: Update Leafelt.js
The is a new version for Leaflet [1.8](https://github.com/Leaflet/Leaflet/releases/tag/v1.8.0)

From the changelog, there should be no breaking changes for us.

### PR DESCRIPTION
**Pull request by Soare-Robert-Daniel**
Fixes #877 

----

### Testing

This is just a dependency update. Just check if everything is alright. Make sure to test it works when upgrading from the older version.

Query

```javascript
new QueryQA().select('blocks').where({ names: ['leaflet-map'] }).run()
```

### CODE REVIEW

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani, regarding the Leaflet plugin for the gestures. It is better to make our modification to avoid that loading error. 

We need to add a timeout that checks if Leaflet is loaded. But this will require compiling the file and therefore integrating it into webpack. Thus, become a part of Otter.

What do you think?

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Yea, that'll be fine. Just make sure to check if that it won't cause any conflicts if a third-party plugin starts enqueuing the same script.

File: inc/class-registration.php [hunk 1]
```diff
@@ -164,7 +164,7 @@ public function enqueue_assets() {
 		$asset_file = include OTTER_BLOCKS_PATH . '/build/blocks/leaflet-map.asset.php';
 		wp_register_script( 'leaflet', OTTER_BLOCKS_URL . 'assets/leaflet/leaflet.js', [], $asset_file['version'], true );
 		wp_script_add_data( 'leaflet', 'async', true );
-		wp_register_script( 'leaflet-gesture-handling', OTTER_BLOCKS_URL . 'assets/leaflet/leaflet-gesture-handling.min.js', array( 'leaflet' ), $asset_file['version'], true );
+		wp_register_script( 'leaflet-gesture-handling', OTTER_BLOCKS_URL . 'build/assets/leaflet-gesture-handling.min.js', array( 'leaflet' ), $asset_file['version'], true );
```
Comment (Soare-Robert-Daniel) [inline inc/class-registration.php:167]: @HardeepAsrani, any idea about that detection if someone loads the same script again with another plugin? This script will load only if someone uses the Maps blocks.

**Comment by irinelenache** [comment]
@Soare-Robert-Daniel Tested the update and everything works fine 🚀

File: inc/class-registration.php (hunk 1 above)
Comment (HardeepAsrani) [reply inc/class-registration.php:167]: @Soare-Robert-Daniel You don't need a detection, just try to load the original script again on the page as see if that's causing any conflicts. If it isn't then it's fine.

Comment (HardeepAsrani) [reply inc/class-registration.php:167]: Anyway, I'll merge it for now. We can leave it for when it happens as that's not the biggest issue.

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Why a separate build for the file? Can't we just make it a part of https://github.com/Codeinwp/otter-blocks/blob/master/webpack.config.js#L90?

**Comment by Soare-Robert-Daniel** [comment]
@HardeepAsrani, I put it separately so that it can be a distinction. I will merge the config if the separation is redundant.

**Comment by HardeepAsrani** [comment]
@Soare-Robert-Daniel Seems fine, you can move the build files from the `assets` folder to `src` folder as we ship the assets folder in the production field, no need to have it shipped.

**Pull request merged**